notifications/pkg/service/service.go

```diff
@@ -0,0 +1,62 @@
+package service
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/cs3org/reva/pkg/events"
+	"github.com/owncloud/ocis/notifications/pkg/channels"
+	"github.com/owncloud/ocis/ocis-pkg/log"
+)
+
+type Service interface {
+	Run() error
+}
+
+func NewEventsNotifier(events <-chan interface{}, channel channels.Channel, logger log.Logger) Service {
+	return eventsNotifier{
+		logger:  logger,
+		channel: channel,
+		events:  events,
+		signals: make(chan os.Signal, 1),
+	}
+}
+
+type eventsNotifier struct {
+	logger  log.Logger
+	channel channels.Channel
+	events  <-chan interface{}
+	signals chan os.Signal
+}
+
+func (s eventsNotifier) Run() error {
+	signal.Notify(s.signals, syscall.SIGINT, syscall.SIGTERM)
+	s.logger.Debug().
+		Msg("eventsNotifier started")
+	for {
+		select {
+		case evt := <-s.events:
+			go func() {
+				switch e := evt.(type) {
+				case events.ShareCreated:
+					msg := "You got a share!"
+					var err error
+					if e.GranteeUserID != nil {
+						err = s.channel.SendMessage([]string{e.GranteeUserID.OpaqueId}, msg)
+					} else if e.GranteeGroupID != nil {
+						err = s.channel.SendMessageToGroup(e.GranteeGroupID, msg)
+					}
+					s.logger.Error().
+						Err(err).
+						Str("event", "ShareCreated").
+						Msg("failed to send a message")
```

Comment:
Wouldn't this be logged in case `err == nil` also?

Reply:
Yes, you're right. Good catch! :+1: